controllers/networkfence_controller.go

```diff
@@ -211,3 +211,40 @@ func (r *NetworkFenceReconciler) unfenceClusterNetwork(ctx context.Context, nwFe
 
 	return nil
 }
+
+// addFinalizerToNetworkFence adds a finalizer to the Networkfence instance.
+func (r *NetworkFenceReconciler) addFinalizerToNetworkFence(
+	ctx context.Context,
+	logger *logr.Logger,
+	networkFence *csiaddonsv1alpha1.NetworkFence) error {
+
+	if !util.ContainsInSlice(networkFence.Finalizers, networkFenceFinalizer) {
+		logger.Info("adding finalizer to NetworkFence object", "Finalizer", networkFenceFinalizer)
+
+		networkFence.Finalizers = append(networkFence.Finalizers, networkFenceFinalizer)
+		if err := r.Client.Update(ctx, networkFence); err != nil {
+			return fmt.Errorf("failed to add finalizer (%s) to NetworkFence resource"+
+				" (/%s) %w", networkFenceFinalizer, networkFence.GetName(), err)
```

Comment:
the format of the string is a little weird:
```go
fmt.Errorf("failed to add finalizer (%s) to NetworkFence resource (/%s) %w"
```
Add a `:` before the `%w`:
```go
fmt.Errorf("failed to add finalizer (%s) to NetworkFence resource (%s): %w"
```

Or instead of `%s`, you can use `%q` to have the strings quoted automatically.